teuthology/provision/pelagos.py

```diff
@@ -0,0 +1,162 @@
+import json
+import logging
+import requests
+import socket
+import time
+import shutil
+import urllib2
+from urllib2 import urlopen, HTTPError
+import re
+
+import teuthology.orchestra
+from teuthology.config import config
+from teuthology.contextutil import safe_while
+from teuthology.exceptions import MaxWhileTries
+from teuthology import misc
+
+log = logging.getLogger(__name__)
+config_section = 'pelagos'
+
+
+def enabled(warn=False):
+    """
+    Check for required Pelagos settings
+
+    :param warn: Whether or not to log a message containing unset parameters
+    :returns: True if they are present; False if they are not
+    """
+    conf = config.get(config_section, dict())
+    params = ['endpoint', 'machine_types']
+    unset = [param for param in params if not conf.get(param)]
```

Comment:
Could you change "param" to "_" for readability, too many 'param-param-pa-pams' in one line :-)